packages/compass-components/src/components/icon-badge.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { spacing } from '@leafygreen-ui/tokens';
+import { css } from '@leafygreen-ui/emotion';
+import { Variant } from '@leafygreen-ui/badge';
+
+import { Badge, Icon } from './leafygreen';
+
+const iconButtonStyles = css({
+  padding: 0,
+  background: 'transparent',
+  border: 'none',
+  lineHeight: 0,
+  cursor: 'pointer',
+});
+
+const contentStyles = css({
+  display: 'flex',
+  alignItems: 'center',
+  gap: spacing[1],
+});
+
+type IconBadgeProps = {
+  text: string;
+  icon: string;
+  dataTestId?: string;
+  onClick?: () => void;
```

Comment:
What's the use-case of having an optional onClick if this component always renders a button?